File: FinanceApp.Backend/FinanceApp.Application/Transactions/Queries/GetAll/TransactionDto.cs
using FinanceApp.Application.Accounts.Queries.GetAllAccounts;
using FinanceApp.Application.Categories.Queries.GetAllCategories;
using FinanceApp.Domain.Enums;

namespace FinanceApp.Application.Transactions.Queries.GetAll;

public class TransactionDto
{
    public Guid Id { get; set; }
    public decimal Amount { get; set; }
    public string Description { get; set; } = null!;
    public TransactionType Type { get; set; }
    public DateTime Date { get; set; }
    
    public CategoryDto Category { get; set; } = null!;
    public AccountDto Account { get; set; } = null!;
}

File: financeapp.frontend/src/app/models/Transactions/TransactionDto.ts
import { Guid } from "guid-typescript";
import { AccountDto } from "../Accounts/AccountDto";
import { CategoryDto } from "../Categories/CategoryDto";
import { TransactionType } from "./TransactionType";

export interface TransactionDto {
    id: string;
    amount: number;
    description: string;
    category: CategoryDto | null;
    account: AccountDto | null;
    type: TransactionType;
    date: Date;        
}
